waku/waku_archive/driver/postgres_driver/postgres_driver.nim

```diff
@@ -853,6 +853,40 @@ proc sleep*(
 
   return ok()
 
+proc acquireDatabaseLock*(
+    s: PostgresDriver, lockId: int = 841886
+): Future[ArchiveDriverResult[void]] {.async.} =
+  ## Acquire an advisory lock (useful to avoid more than one application running migrations at the same time)
+  let lockedRes = await s.getStr(
+    fmt"""
+    SELECT pg_try_advisory_lock({lockId})
+    """
+  )
+  if lockedRes.isErr():
+    return err("error acquiring a lock: " & $lockedRes.error)
+
+  if lockedRes.value == "f":
+    return err("another waku instance is currently executing a migration")
```

Comment:
I think it could be a little bit more concise with the `valueOr` approach ( notice that we also use `isOkOr` for `Result[void, ...]` )
```suggestion
  let locked = (await s.getStr(
    fmt"""
    SELECT pg_try_advisory_lock({lockId})
    """
  )).valueOr:
    return err("error acquiring a lock: " & $error)

  if locked == "f":
    return err("another waku instance is currently executing a migration")
```